### ISSUE TITLE
Entity References

### ISSUE THREAD

**Issue by CsarRamos** [systopia/de.systopia.resource#27]
Hello, 

Using searchkit, it seems that the resource entities are not linked to the core entities. have you detected this? Maybe searchkit can't handle these joins, I see this is because DAO files are using dynamic foreign keys. I'm not sure if this would be a new enhancement to searchkit or is not posible.

Thanks,

**Comment by bjendres** [comment]
Yes, this is a very loose link as it allows linking to any other entity. 

This probably "only" needs to be expressed differently in the XML/DAO, for APIv4 to pick it up correctly.

Any chance you could provide a PR?

**Comment by Fabian-SYSTOPIA** [comment]
[There are plans to create an custom "Entity Reference" field type.](https://lab.civicrm.org/dev/core/-/issues/3721#note_85459)
This field could then be used to link custom entities to other (core) entities and use these links in CiviCRMs search kit. 

By the way even know you should already be able to create a custom contact reference field for a custom entity and use the link to the contact's data in search kit. This may however not work for all purposes but I have used it to create custom formss (using form builder) and show custom entities in a contact view using the contact layout editor.

**Comment by CsarRamos** [comment]
Thanks @bjendres @Fabian-SYSTOPIA, it sounds like an interesting approach that would improve the funcionality considerably. We would like to work with the event class in search kit and resoure demands, I have created a pr to propose a change in the demand class to give the same functionality as the resource class. https://github.com/systopia/de.systopia.resource/pull/28

**Comment by jensschuppe** [comment]
@CsarRamos would you be able to provide a PR for the `Resource` entity as well, now that #28 is merged for the `ResourceDemand` entity?

**Comment by jensschuppe** [comment]
Reopening for implementation for `Resource` class.

**Comment by CsarRamos** [comment]
> @CsarRamos would you be able to provide a PR for the `Resource` entity as well, now that #28 is merged for the `ResourceDemand` entity?

Hi, great, thank you for the merge. I will check it  and try to adapt Resource class following the same approach, I will let you know when I have news, cheers.
